base/sparse/sparsematrix.jl

```diff
@@ -1428,7 +1433,7 @@ end
 
 speye_scaled(diag, m::Integer, n::Integer) = speye_scaled(typeof(diag), diag, m, n)
 
-function speye_scaled(T, diag, m::Integer, n::Integer)
+function speye_scaled(::Type{T}, diag, m::Integer, n::Integer) where T
```

Comment:
this is technically more restrictive than the existing signature, but looks like no one is calling this in registered packages and all the call sites in base pass a type